testsuite/features/step_definitions/command_steps.rb

```diff
@@ -863,10 +863,18 @@
 
 When(/^I (enable|disable) Debian-like "([^"]*)" repository on "([^"]*)"$/) do |action, repo, host|
   node = get_target(host)
-  raise ScriptError, "#{node.hostname} is not a Debian-like host." unless deb_host?(host)
 
-  source_repo = "deb http://archive.ubuntu.com/ubuntu/ $(lsb_release -sc) #{repo}"
-  node.run("sudo add-apt-repository -y -u #{action == 'disable' ? '--remove' : ''} \"#{source_repo}\"")
+  # add-apt-repository does not work with new deb822 format
+  file = 'edit-deb822.awk'
+  source = "#{File.dirname(__FILE__)}/../upload_files/#{file}"
+  dest = "/tmp/#{file}"
+  success = file_inject(node, source, dest)
+  raise ScriptError, 'File injection failed' unless success
+
+  # edit ubuntu.sources with downloaded utility
+  sources = '/etc/apt/sources.list.d/ubuntu.sources'
```

Comment:
Can we parametrize `ubuntu` word?
I mean, the Gherkin step says Debian-like....